src/config/model.rs

```diff
@@ -382,13 +400,21 @@ impl Config {
     }
 
     pub fn signature(&self, account: &Account) -> Option<String> {
+        let default_sig_delim = String::from("-- \n");
+        let sig_delim = account
+            .signature_delimiter
+            .as_ref()
+            .or_else(|| self.signature_delimiter.as_ref())
+            .unwrap_or(&default_sig_delim);
         let sig = account
             .signature
             .as_ref()
             .or_else(|| self.signature.as_ref());
-
-        sig.and_then(|sig| fs::read_to_string(sig).ok())
+        sig.and_then(|sig| shellexpand::full(sig).ok())
+            .map(|sig| sig.to_string())
+            .and_then(|sig| fs::read_to_string(sig).ok())
             .or_else(|| sig.map(|sig| sig.to_owned()))
+            .map(|sig| String::new() + sig_delim + sig.as_ref())
```

Comment:
Hm... wouldn't a `concat![sig_delim, sig.as_ref()]` suit here?

Reply:
Only literals can be passed to `concat!`. But I've found a more readable way: `sig_delim.to_owned() + sig.as_ref()`. Thanks for pointing it out!